util/exponential-fallback.js

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const MAX_RETRIES = 5
+const MAX_WAIT_INTERVAL_MS = 1 * 60 * 1000 // 1 minute in milliseconds
+const RETRY_COUNT_MULTIPLIER_MS = 100
+
+const logger = require('@adobe/aio-lib-core-logging')('@adobe/aio-lib-core-tvm', { provider: 'debug' })
+const fetch = require('node-fetch').default
+const util = require('util')
+const sleep = util.promisify(setTimeout)
+
+/**
+ * This function will retry connecting to a url end-point, with
+exponential fallback. Returns a Promise.
+ *
+ * @param {string} url endpoint url
+ * @param {object} options request options
+ * @param {number} retryCount retry count
+ * @param {object} err error response
+ * @returns {Promise} Promise object representing the http response
+ */
+async function exponentialFallback (url, options, retryCount, err) {
```

Comment:
As we will likely flesh this function out and reuse it across the sdk, I would change some things in the interface to make it more generic/cleaner:

- wdyt about having `exponentialBackoff (url, requestOptions, retryOptions)` ?

- rename exponentialFallback to Backoff ? (a google search gave me results for backoff)
- pass the retries, interval, and multiplier as `retryOptions` with defaults set as **{ retries: 5, multiplier: 1.5, interval: 500 }**
- remove the retryCount
- remove the error option as this is not part of the interface (set by helper)
    - to make it easier you can implement recursion more like
```
async function exponentialBackoff (url, requestOptions, retryOptions) {
  // check params + set defaults
  return _exponentialBackoffHelper (url, requestOptions, retryOptions)
}
async function _exponentialBackoffHelper (url, requestOptions, retryOptions) {
  // decrease retryOptions.retries
  // await make the request
  // if error && retries != 0
    // adjust retryOptions.interval based on Backoff formula
    // wait interval
    // call  _exponentialBackoffHelper
  // else return response (which can be error response or good response)
}
```